### PR TITLE
feat: liquidator query enabled by runtime flag

### PR DESCRIPTION
**Pull request by toteki**
## Description

The query `liquidation-targets` is enabled at runtime the node is started with a flag:

```bash
# Enabled
umeed start --enable-liquidator-query

# Enabled
umeed start -l

# Disabled
umeed start
```

closes: #1400

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/leverage/types/flag.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package types
+
+const FlagEnableLiquidatorQuery = "enable-liquidator-query"
+
+// EnableLiquidator must be set to true to enable QueryLiquidationTargets
+var EnableLiquidator *bool
```
Comment (robert-zaremba) [inline x/leverage/types/flag.go:6]: ouch. A global, public pointer is a something we should definitely avoid.
Why not adding this flag to app cmd, and passing it through `AppModule` down to the keeper?
Maybe here: https://github.com/umee-network/umee/blob/main/cmd/umeed/cmd/root.go#L188

**Review by robert-zaremba — CHANGES_REQUESTED** [review]
Let's avoid the global variables, especially if we modify them in tests, and even more especially if they are pointers.

File: x/leverage/types/flag.go (hunk 1 above)
Comment (toteki) [reply x/leverage/types/flag.go]: Is there a way to get from `leverage.AddModuleInitFlags(startCmd)` to the keeper constructor? If so, I can place the flag at `keeper.EnableLiquidatorQuery`

Comment (robert-zaremba) [reply x/leverage/types/flag.go]: Through `appOpts`. Check here: https://github.com/umee-network/umee/blob/main/cmd/umeed/cmd/app_creator.go#L53

Comment (robert-zaremba) [reply x/leverage/types/flag.go]: You have also access to it in `New` in app.go.

File: app/app.go [hunk 2]
```diff
@@ -442,13 +442,16 @@ func New(
 		distrtypes.ModuleName,
 	)
 	var err error
+	// TODO: enable query on flag. where does this come from?
```
Comment (toteki) [inline app/app.go:445]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -47,8 +47,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ## [Unreleased]
 
 ### Features
-
 - [1555](https://github.com/umee-network/umee/pull/1555) Updates IBC to v5.1.0 that adds adds optional memo field to `FungibleTokenPacketData` and `MsgTransfer`.
+- [1577](https://github.com/umee-network/umee/pull/1577) Removes LIQUIDATOR build flag and adds `--enable-liquidator-query` or `-l` runtime flag to `umeed start`.
```
Comment (robert-zaremba) [inline CHANGELOG.md:51]:
```suggestion
- [1577](https://github.com/umee-network/umee/pull/1577) Removes LIQUIDATOR build flag and adds `--enable-liquidator-query` or `-l` runtime flag to `umeed start`. See [README.md](README.md) file for more details.
```

File: app/app.go [hunk 4]
```diff
@@ -442,13 +442,15 @@ func New(
...
 	var err error
+	enableLiquidatorQuery := cast.ToBool(appOpts.Get(leveragetypes.FlagEnableLiquidatorQuery))
 	app.LeverageKeeper, err = leveragekeeper.NewKeeper(
 		appCodec,
 		keys[leveragetypes.ModuleName],
 		app.GetSubspace(leveragetypes.ModuleName),
 		app.BankKeeper,
 		app.OracleKeeper,
 		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		enableLiquidatorQuery,
```
Comment (robert-zaremba) [inline app/app.go:453]:
```suggestion
	app.LeverageKeeper, err = leveragekeeper.NewKeeper(
		appCodec,
		keys[leveragetypes.ModuleName],
		app.GetSubspace(leveragetypes.ModuleName),
		app.BankKeeper,
		app.OracleKeeper,
		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
		cast.ToBool(appOpts.Get(leveragetypes.FlagEnableLiquidatorQuery)),
```

**Review by robert-zaremba — APPROVED** [review]
thanks. LGTM.

**Pull request merged**